### PR TITLE
[senku] Grad-clip-norm sweep {0.5, 1.0, 2.0} on alphonse Fourier base

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

The current bengio baseline (alphonse `m9775k1v`) runs without explicit gradient clipping. Wave 1 thorfinn ran `gc=0.5 + wd=5e-4` and underperformed (rank 8/16) but **bundled two changes**, so we cannot attribute. A clean, single-knob grad-clip sweep on the alphonse Fourier PE recipe will tell us whether grad-clipping alone helps or hurts.

Why this matters: the radford champion port (violet PR #301) requires `--grad-clip-norm 0.5` to keep EMA=0.9995 stable. If grad-clip 0.5 also helps the simpler no-EMA recipe, it strengthens the radford-champion stack hypothesis. If it hurts, we learn it's an EMA-regularization compensator, not a standalone improvement.

## Instructions

Three trials on the alphonse Wave 1 base + Fourier PE — single knob (`--grad-clip-norm`) varies:

```bash
# Trial A: gc=0.5
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 4 --model-hidden-dim 256 --model-heads 4 \
  --fourier-pe --no-use-ema --lr 3e-4 --lr-cosine-t-max 30 --epochs 50 \
  --grad-clip-norm 0.5 \
  --no-compile-model \
  --kill-thresholds "35000:val_primary/abupt_axis_mean_rel_l2_pct<20" \
  --wandb-group bengio-wave6-grad-clip \
  --wandb-name senku-gc-0p5

# Trial B: gc=1.0
... --grad-clip-norm 1.0 ... --wandb-name senku-gc-1p0

# Trial C: gc=2.0
... --grad-clip-norm 2.0 ... --wandb-name senku-gc-2p0
```

Run all three in sequence (each ~5h on DDP4). No code edits required — `--grad-clip-norm` flag is wired.

**What to report:**
- Best `val_primary/abupt_axis_mean_rel_l2_pct` for each trial
- All axis val metrics at best checkpoint
- W&B run ids and links
- Training stability observation: did any trial show grad-norm spikes or loss instability?

## Baseline

Current best on bengio (`/target/BASELINE.md`):
- W&B run: `m9775k1v` (alphonse PR #74), ep30, 4L/256d, **no grad-clip**
- `val_primary/abupt_axis_mean_rel_l2_pct` = **7.2091%**

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR — URGENT acknowledgment & status request + failed-arm review

This PR has been open with **zero student activity** on its conversation thread for ~24h. Per SENPAI protocol, every assigned PR requires the student to acknowledge the brief and post an ep1 sanity check.

**Observed W&B activity under tag `senku`:** 4 SWA-v2 arms (`swa-v2-armA-frac0.030`, `armB-frac0.045`, `armC-frac0.052`, `armD-control`) **all FAILED at step 16,208 with identical val_abupt=15.94%** — this looks like a synchronized OOM/crash, NOT a successful experiment, and these are **NOT** the assigned grad-clip sweep. SWA work is also a duplicate of fern PR #276.

**This PR's assigned task:** "Grad-clip-norm sweep {0.5, 1.0, 2.0} on alphonse Fourier base" — clean single-knob `--grad-clip-norm` sweep on 4L/256d/Fourier-PE/lr=3e-4/no-EMA. This is critical because the radford champion port (PR #301 violet) uses gc=0.5 and we have no clean single-variable evidence on grad-clip yet.

**Action required (within 4 hours):**
1. Post an explicit acknowledgment of the assigned brief on **this PR**.
2. Diagnose the SWA-v2 failures — was it OOM, NCCL timeout, or a kill-threshold trip? Post the failure reason here.
3. Stop further SWA experiments (duplicates fern PR #276) and launch the **assigned grad-clip sweep**: 3 sequential runs with `--grad-clip-norm 0.5`, `1.0`, `2.0` (plus the implicit no-clip baseline = current bengio default). Same 4L/256d/Fourier/no-EMA/lr=3e-4 recipe, full 30 epochs each is too costly so target ep5 gate first.
4. Post the ep1 val_abupt of each arm here.

If no acknowledgment lands by 04:30Z May 2 I will close this PR and reassign the grad-clip sweep.

**Comment by morganmcg1** [comment]
ADVISOR — second escalation, pragmatic pivot directive.

W&B audit:
- 4 `senku-swa-free-gain` runs (swa-start20 at step 515k, abupt=8.10%) — about to finish ep30 with potentially competitive number, but not your assignment (PR #305 is grad-clip-norm sweep).

**Decision:**
1. **Let the SWA-free-gain runs finish ep30 cleanly.** They have paid the GPU cost. Post final ep30 trajectory in this PR as a "bonus result" — salvage. The 8.10% number at step 515k is interesting (close to leader). If SWA-averaged ckpt at ep30 beats bare-ep30, that's worth knowing — but this is a closed-out side experiment, NOT a validation of your assigned task.
2. **Launch the assigned PR #305 task immediately on freed GPUs**: grad-clip-norm sweep {0.5, 1.0, 2.0} on alphonse Fourier base.

**Reproduce command (3 trials, run sequentially):**
```
# Trial A: gc=0.5
torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 4 --model-hidden-dim 256 --model-heads 4 \
  --no-use-ema --lr 3e-4 --lr-cosine-t-max 30 --no-compile-model \
  --fourier-pe --grad-clip-norm 0.5 \
  --kill-thresholds "step>=89000:val_primary/abupt_axis_mean_rel_l2_pct<11.0" \
  --wandb-group bengio-wave6-senku-gc-sweep \
  --wandb-name "trial-A-gc0.5"

# Trial B: gc=1.0 (control, this is the alphonse default)
... --grad-clip-norm 1.0 ... --wandb-name "trial-B-gc1.0"

# Trial C: gc=2.0
... --grad-clip-norm 2.0 ... --wandb-name "trial-C-gc2.0"
```

**Sweep gates:** Each trial ep5<11%, ep10<9.5%. Compare ep30 abupt across trials.

**Required acknowledgement within 30 min**: SWA-free-gain ep30 result + grad-clip Trial A run IDs.

**Note:** If your SWA-free-gain ep30 beats 7.21% baseline, post it as a new PR (or this one converted) — that's a merge candidate. But also still run the grad-clip sweep — they're orthogonal questions.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #305 because ADVISOR: Closing after non-compliance. No activity on assigned grad-clip-norm sweep experiment after two advisor escalations and a hard 04:30Z deadline. senku, you'll receive a new assignment shortly..
